Observer/AfterConfigSaveObserver.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace Razorpay\Magento\Observer;
+
+// require in case of zip installation without composer
+require_once __DIR__ . "/../../Razorpay/Razorpay.php";
+
+use Razorpay\Api\Api;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Sales\Model\Order\Payment;
+use Razorpay\Magento\Model\PaymentMethod;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\App\Config\Storage\WriterInterface;
+use Magento\Framework\App\RequestInterface;
+use Razorpay\Magento\Model\Config;
+
+
+/**
+ * Class AfterConfigSaveObserver
+ * @package Razorpay\Magento\Observer
+ */
+class AfterConfigSaveObserver implements ObserverInterface
+{
+
+    /**
+     * Store key
+     */
+    const STORE = 'store';
+
+
+    private $request;
+    private $configWriter;
+
+    /**
+     * StatusAssignObserver constructor.
+     *
+     * @param OrderRepositoryInterface $orderRepository
+     */
+    public function __construct(
+        \Razorpay\Magento\Model\Config $config,
+        RequestInterface $request, 
+        WriterInterface $configWriter,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Psr\Log\LoggerInterface $logger
+    ) {
+        $this->config = $config;
+        $this->request = $request;
+        $this->configWriter = $configWriter;
+        $this->_storeManager = $storeManager;
+        $this->logger          = $logger;
+        
+        $this->config = $config;
+
+        $this->key_id = $this->config->getConfigData(Config::KEY_PUBLIC_KEY);
+        $this->key_secret = $this->config->getConfigData(Config::KEY_PRIVATE_KEY);
+
+        $this->rzp = new Api($this->key_id, $this->key_secret);
+
+        $this->webhookUrl = $this->_storeManager->getStore()->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_WEB) . 'razorpay/payment/webhook';
+
+        $this->webhookId = null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(Observer $observer)
+    { 
+
+        $razorpayParams = $this->request->getParam('groups')['razorpay']['fields'];
+        
+        try
+        {
+            $webhookPresent = $this->getExistingWebhook();
+
+            if(empty($razorpayParams['enable_webhook']['value']) === true)
+            {
+                $this->disableWebhook();
+                return;
+            }
+
+        
+            if(in_array($_SERVER['SERVER_ADDR'], ["127.0.0.1","::1"]))
```

Comment:
@ramth05 Can we do this check before calling any webhook API?

Reply:
done